### ISSUE TITLE
Unexpected token for `as` operator

### ISSUE THREAD

**Issue by bondom** [eslint/typescript-eslint-parser#455]
**What version of TypeScript are you using?**
2.7.1

**What version of `typescript-eslint-parser` are you using?**
14.0.0

**What code were you trying to parse?**

```ts
const a:number = 5;
console.log(a as any);
```

**What did you expect to happen?**
I expect to see nothing in console when running 
`eslint --parser typescript-eslint-parser ./src/index.tsx`

**What happened?**

```
SyntaxError: Unexpected token, expected "," (18:15)
In console I see:
  16 | // );
  17 | const a:number = 5;
> 18 | console.log(a as any);
     |               ^
```

Maybe I missed something..

**Comment by bondom** [comment]
This error was caused by next lines in my `.eslintrc` file:
```
"prettier/prettier": [
      "error",
      {
        "singleQuote": true
      }
    ]
```

**Comment by bondom** [comment]
I decided to reopen issue because I am interested why it doesn't work. Can someone please explain why it fails with these lines?

**Comment by bondom** [comment]
I found cause of this issue! I didn't specify `parser` in prettier config. So, I added it and everything works fine:
```
 "prettier/prettier": [
      "error",
      {
        "singleQuote": true,
        "parser": "typescript"
      }
    ]
```
